### DISCUSSION TITLE
UI Support for promiscuous mode

### DISCUSSION THREAD

**Discussion by fdizazzo** [eclipse-kura/kura#5068] [Ideas]
Hi all!
Opening this as a follow up to #4957, trying to understand the best way to integrate the new functionalities.
I see two main points:
- Let the user see: NetworkManager version, with MTU for IPv6 and promiscuous mode requiring a minimum version; see the status of promiscuous mode, as the status returned from the device driver (either as a specific field or as a list of all flags active on an interface).
Below an integration proposal on the currently existing Hardware tab:
![tab_hardware](https://github.com/eclipse/kura/assets/55892769/74d110ea-6933-4478-82a4-b14bdbbbfa10)

- Let the user configure the promiscuous mode: this could be in a new 'Advanced' tab, either on its own or along the newly introduced MTU configuration fields, or... somewhere else for which I'm accepting suggestions 😄 
![tab_advanced](https://github.com/eclipse/kura/assets/55892769/a6e55e86-cefb-4ac4-bcfb-1e8bcefb028e)

Thanks for your support! 🙏

**Comment by mattdibi** [comment]
Hello there 👋 @fdizazzo,

thanks for reaching out.

I would involve both @MMaiero and @pierantoniomerlino in the discussion. @MMaiero do we want to display the NetworkManager version in our status? If we do for NM should we do the same for MM?

The idea here is that there's an increasing number of features that are tied to the underlying NM/MM version, therefore the user might be interested in knowing that.

**Comment by MMaiero** [comment]
We already have the packages versions in the inventory. Would't that imply a bit of duplication?

**Reply by mattdibi** [reply]
You’re referring to https://eclipse.github.io/kura/docs-release-5.4/administration/system-component-inventory/ ? I didn’t think about it. @fdizazzo what do you think? Would that be enough to inform the user?

**Reply by fdizazzo** [reply]
Completely missed that 👼 
Looks good, the main difference I see is that inventory brings up the version advertised by the package while I was thinking at the one retrieved via DBus api, but I guess it would hardly make any difference in a real world scenario.

**Comment by mattdibi** [comment]
Ok so... we agreed that we already have a solution in place for retrieving the NetworkManager version.

@MMaiero what about the "Advanced" tab? 

This was a thing we talked about in the past and was not justified since we only had the MTU parameter. We now have the "Promiscuous mode" that would fit in that tab.

The drawback of having this new tab is the fact that we would need to have **two** MTUs setting differentiated by IP version (as per @fdizazzo excellent mock-up) while now, they live in their respective IP version tab (which makes sense).

I'm having a hard time of coming up with alternatives here since the "Promiscuous mode" is a property of **the interface**... in our NetworkManager implementation it corresponds to the [`accept-all-mac-addresses` property](https://networkmanager.dev/docs/api/latest/settings-802-3-ethernet.html) which belongs to the 802-3-ethernet settings. We don't have a direct mapping in our UI (if it belonged to the IPv4 settings we would be able to map it in a simple way to the already existing tab).

I'm open to suggestions.

**Comment by MMaiero** [comment]
I think that now we may have enough parameters to give some dignity to the "Advanced" tab (even though I am not fully convinced with the advanced label). The MTU param is a "pro feature" and it makes more sense to be moved to a "pro" tab along with the promiscuous configuration. So I agree with the @fdizazzo mockup and think it is the cleanest approach for an end user distinguishing between common and basic features compared to advanced ones.

**Comment by mattdibi** [comment]
Addressed in https://github.com/eclipse/kura/pull/5087

@fdizazzo can you take care of the documentation update too?

**Reply by fdizazzo** [reply]
Sure, i'll do in the next few days 👍

**Comment by mattdibi** [comment]
Documentation addressed in https://github.com/eclipse/kura/pull/5104

Thanks @fdizazzo
